modules/host/negotiate.go

```diff
@@ -234,7 +228,7 @@ func (h *Host) rpcContract(conn net.Conn) (err error) {
 	}
 
 	// Add the collateral to the transaction, but do not sign the transaction.
-	collateralTxn, txnID, err := h.addCollateral(unsignedTxn, terms)
+	collateralTxn, txnBuilder, err := h.addCollateral(unsignedTxn, terms)
```

Comment:
would probably be cleaner if you created the TransactionBuilder before, and passed it `addCollateral` for modification.